soda/scientific/soda/scientific/distribution/comparison.py

```diff
@@ -99,27 +110,22 @@ def _parse_reference_cfg(self, ref_file_path: FilePath) -> RefDataCfg:
         return RefDataCfg.parse_obj(ref_data_cfg)
 
 
-class DistributionAlgorihm:
+class DistributionAlgorithm(abc.ABC):
     def __init__(self, cfg: RefDataCfg, test_data: pd.Series, seed: int = 61) -> None:
-        self.cfg = cfg
         self.test_data = test_data
-        self.rng = np.random.default_rng(seed)
-        self.ref_data = self.generate_ref_data()
-
-    @abc.abstractmethod
-    def evaluate(self, test_data: pd.Series) -> Tuple[float, float]:
-        ...
+        self.ref_data = generate_ref_data(cfg, len(test_data), np.random.default_rng(seed))
 
     @abc.abstractmethod
-    def generate_ref_data(self) -> pd.Series:
+    def evaluate(self) -> dict:
         ...
 
 
-class ChiSqAlgorithm(DistributionAlgorihm):
+class ChiSqAlgorithm(DistributionAlgorithm):
     def __init__(self, cfg: RefDataCfg, test_data: pd.Series, seed: int = 61) -> None:
+        cfg.dtype = "categorical"
```

Comment:
Sorry my bad this is something from a previous approach that shouldn't be there. My initial approach was to create a `dtype` that would be set using the `generate_ref_data` method, depending on whether the data was of categorical or continuous nature. This method would decide whether the data was categorical or continuous and would not require user input and would always be set to categorical for Chi-square and always be set to continuous for KS. However, Baturay told me to drop that approach since it's not easy to determine whether data is continuous or categorical.